rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/TreeView.java

```diff
@@ -253,17 +225,28 @@ public void onClick(MenuItemClickEvent event) {
                     }
                 }
 
-                if (currentSelectedNodeSource != null &&
-                    currentSelectedNodeSource.getNodeSourceStatus().equals(NodeSourceStatus.NODES_UNDEPLOYED)) {
-                    deployItem.setEnabled(true);
+                if (currentSelectedNodeSource != null) {
+                    switch (currentSelectedNodeSource.getNodeSourceStatus()) {
+                        case NODES_DEPLOYED:
+                            deployItem.setEnabled(false);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Make this anonymous inner class a lambda [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1604)